### PR TITLE
FlexPRET platform support

### PR DESCRIPTION
**Pull request by lsk567**
This PR uses a build script to build LF programs for FlexPRET. This approach is borrowed from prior examples on nRF52+Buckler and XMOS.

### CODE REVIEW

**Comment by lhstrh** [comment]
This PR doesn't look right to me. It copies sources from other repositories and checks in generated sources as well.

**Comment by lhstrh** [comment]
Oh, wait, those files are getting _deleted_. Then it looks right, but makes me wonder what was happening in this repo up until now...

File: platform/include/encoding.h [hunk 1]
```diff
@@ -0,0 +1,2832 @@
+/* See LICENSE for license details. */
+
+#ifndef RISCV_CSR_ENCODING_H
+#define RISCV_CSR_ENCODING_H
+
```
Comment (lhstrh) [inline platform/include/encoding.h:5]: Comments missing

**Review by lhstrh — COMMENTED** [review]
Upon taking a closer look, this seems like great progress. I realize this wasn't yet ready for review, but it could use some cleanup and comments here and there.

File: platform/include/encoding.h (hunk 1 above)
Comment (lsk567) [reply platform/include/encoding.h:5]: Thanks for the catch. Will focus more on documentation moving forward...

**Pull request merged**